ext/mbstring/mbstring.c

```diff
@@ -2183,6 +2190,202 @@ PHP_FUNCTION(mb_output_handler)
+    zend_long split_length = 1;
+
+    ZEND_PARSE_PARAMETERS_START(1, 3)
+            Z_PARAM_STR(str)
+            Z_PARAM_OPTIONAL
+            Z_PARAM_LONG(split_length)
+            Z_PARAM_STR(encoding)
+    ZEND_PARSE_PARAMETERS_END();
+
+    if (split_length <= 0) {
+        php_error_docref(NULL, E_WARNING, "The length of each segment must be greater than zero");
+        RETURN_FALSE;
+    }
+
+    /* fill mbfl_string structure */
+    string.no_language = MBSTRG(language);
+    string.encoding = php_mb_get_encoding(encoding);
+    if (!string.encoding) {
+        RETURN_FALSE;
+    }
+
+    p = (char unsigned *)ZSTR_VAL(str); /* string cursor pointer */
+    string.val = (char unsigned *)p;
+    string.len = ZSTR_LEN(str);
+    last = string.val + string.len; /* last string char pointer */
+
+    /* check if split_length > string.len */
+    if(split_length > string.len){
+        array_init_size(return_value, 1);
+        add_next_index_stringl(return_value, (const char *)p, string.len);
+        return;
+    }
+
+    /*
+     * +----------------------------------------------------------------------+
+     * |                           3 scenarios                                |
+     * +----------------------------------------------------------------------+
+     */
+    mbfl_encoding = string.encoding;
+
+    /*
+     * +----------------------------------------------------------------------+
+     * | first scenario:    1,2,4-bytes fixed width encodings (head part)     |
+     * +----------------------------------------------------------------------+
+     */
+    if (mbfl_encoding->flag & MBFL_ENCTYPE_SBCS) { /* 1 byte */
+        mb_len = string.len;
+        chunk_len = (size_t)split_length; /* chunk length in bytes */
+
+    } else if (mbfl_encoding->flag & (MBFL_ENCTYPE_WCS2BE | MBFL_ENCTYPE_WCS2LE )) { /* 2 bytes */
+        mb_len = string.len / 2;
+        chunk_len = split_length * 2;
+
+    } else if (mbfl_encoding->flag & (MBFL_ENCTYPE_WCS4BE | MBFL_ENCTYPE_WCS4LE )) { /* 4 bytes */
+        mb_len = string.len / 4;
+        chunk_len = split_length * 4;
+    /*
+     * +----------------------------------------------------------------------+
+     * | second scenario: "variable width encodings with length table"        |
+     * +----------------------------------------------------------------------+
+     */
+    } else if (mbfl_encoding->mblen_table != NULL) {
+        char unsigned const *mbtab = mbfl_encoding->mblen_table;
+
+        /* assume that we have 1-bytes characters */
+        array_init_size(return_value, (string.len + split_length) / split_length); /* round up */
+
+        while (p < last) { /* split cycle work until the cursor has reached the last byte */
+            char const *chunk_p = p; /* chunk first byte pointer */
+            chunk_len = 0; /* chunk length in bytes */
+            for (zend_long char_count = 0; char_count < split_length, p <= last; ++char_count) {
```

Comment:
The `,` here should be `&&`, otherwise the first condition is just ignored.

Reply:
Also, shouldn't this be `p < last`?

Reply:
And the `if (p > last)` check below can probably be dropped now.

Reply:
yes, you are right as always. :-)